File: solutions/python/nth-prime/1/nth_prime.py
import math

def prime(number):
    if number < 1:
        raise ValueError("there is no zeroth prime")
    elif number == 1:
        return 2
    # odd prime numbers
    i = 2
    prime = 3
    while i < number:
        prime += 2
        if is_prime(prime):
            i += 1
    return prime

def is_prime(n):
    return not any(map(lambda x: n % x == 0, range(3, int(math.sqrt(n))+1)))

File: solutions/javascript/nth-prime/1/nth-prime.js
export const prime = (n) => {
  if(n === 0) {
    throw new Error("there is no zeroth prime");
  }
  let current = 1;
  let prime = 2;
  while(current < n) {
    prime++;
    if (isPrime(prime)) {
      current++;
    }
  }
  return prime;
};

function isPrime(n) {
  for(let i = 2; i<=Math.ceil(Math.sqrt(n)); i++) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
}